src/test/java/teammates/test/cases/ui/pagedata/InstructorFeedbackResultsPageDataTest.java

```diff
@@ -0,0 +1,1584 @@
+import teammates.ui.template.InstructorFeedbackResultsParticipantPanel;
+import teammates.ui.template.InstructorFeedbackResultsQuestionTable;
+
+public class InstructorFeedbackResultsPageDataTest extends BaseComponentTestCase {
+    private static final String NOT_IN_A_SECTION = "Not in a section";
+
+    private static DataBundle dataBundle = getTypicalDataBundle();
+    
+    // logic is used in this page data test, but not other page data tests, because
+    // unfortunately, logic is required to construct the ResultsBundle
+    Logic logic = new Logic();
+    
+    @BeforeClass
+    public static void classSetUp() throws Exception {
+        printTestClassHeader();
+        removeAndRestoreTypicalDataInDatastore();
+    }
+    
+    @Test
+    public void testInitForViewByQuestion() throws UnauthorizedAccessException, EntityDoesNotExistException {
+        AccountAttributes account = dataBundle.accounts.get("instructor1OfCourse1");
+        InstructorFeedbackResultsPageData data = new InstructorFeedbackResultsPageData(account);
+        
+        InstructorAttributes instructor = dataBundle.instructors.get("instructor1OfCourse1");
+        
+        ______TS("typical case: view all sections, all questions, show stats");
+        
+        String courseId = instructor.courseId;
+        String feedbackSessionName = dataBundle.feedbackSessions.get("session1InCourse1").feedbackSessionName;
+        String showStats = "on";
+        String groupByTeam = "on";
+        String sortType = "question";
+        String selectedSection = InstructorFeedbackResultsPageData.ALL_SECTION_OPTION;
+        
+        data.setBundle(logic.getFeedbackSessionResultsForInstructorWithinRangeFromView(
+                                        feedbackSessionName, courseId, instructor.email, 
+                                        1000, sortType));
+        data.initForViewByQuestion(instructor, selectedSection, showStats, groupByTeam);
+        
+        List<InstructorFeedbackResultsQuestionTable> questionPanels = data.getQuestionPanels();
+        assertEquals(4, questionPanels.size());
+        
+        // Verify that the first question table is correct
+        InstructorFeedbackResultsQuestionTable questionPanel = questionPanels.get(0);
+        verifyQuestionTableForQuestionView(questionPanel, dataBundle.feedbackQuestions.get("qn1InSession1InCourse1").questionMetaData.getValue());
+        
+        List<InstructorFeedbackResultsResponseRow> responseRows = questionPanel.getResponses();
+        assertEquals(5, responseRows.size());
+        // first response
+        InstructorFeedbackResultsResponseRow responseRow = responseRows.get(0);
+        verifyResponseRow(responseRow, dataBundle.feedbackResponses.get("response1ForQ1S1C1"), 
+                                       dataBundle.students.get("student1InCourse1"), 
+                                       dataBundle.students.get("student1InCourse1"));
```

Comment:
Indentation is off at several places, like this where it looks like the third parameter is nested more deeply.